tsdb/index/index.go

```diff
@@ -1852,7 +1852,7 @@ func (dec *Decoder) Series(b []byte, lbls *labels.Labels, chks *[]chunks.Meta) e
 	k = d.Uvarint()
 
 	if k == 0 {
-		return nil
+		return d.Err()
```

Comment:
If an error occurs in the 2 `d.Uvarint()` calls above, `d.Uvarint()` returns 0 and sets an error, but it's never checked. If we return `d.Err()` (like we already do at the end of the function) then in case of error it will be returned back.